### ISSUE TITLE
Improvements to book list : add send to button

### ISSUE THREAD

**Issue by hugoheitz** [seblucas/cops#329]
I created the issue #326 with a lot of different situations. 

I would like to split it to be able to track progress and to explain more precisely things.

It would be great to have the sendto button directly in the book list 
here the last fifty books
https://cloud.githubusercontent.com/assets/972023/23554791/caec990e-0026-11e7-86b9-36ff9601b53a.PNG

or in the case of a search.

For the moment, the sendto button is only available on the book information page.

**Comment by horus68** [comment]
I don't see the merits of it... What would the recipient gain from that? A list of 15 books in a zip?
How would he know that he really wants that book?
Or is just an mail to yourself?

**Comment by hugoheitz** [comment]
No zip.

For each book on the list add the sendto button.

Clear example : you have read a book from a serie and  you add the other
ones to calibre.
On cops you want to access the last books added and easily send the other
books from the serie to the ereader device.

**Comment by horus68** [comment]
ok, so hitting the button mail from the list and not having to enter the details seems ok for me

**Comment by hugoheitz** [comment]
One thing needs to be explained :
In this usecase, the person who browses in cops and the recipient ( the
destination mail ) owns the smartphone and the destination ereader.

**Comment by hugoheitz** [comment]
yes, for each book you can hit the mail button directly and not having to enter the details.

I tried to create a mockup  

![mockup cops list with mail button](https://cloud.githubusercontent.com/assets/972023/23821477/e7cd2b52-0633-11e7-934c-166b5f294728.png)

**Comment by hugoheitz** [comment]
I just bought a new ereader (kindle) supporting the sendto functionality for a member of my family.

Using COPS on a smartphone and the sendto functionality to put books on the selected kindle ereader or app is great BUT with the current implementation, it misses something, you can't select easily where you want to send, you have to provide an email address.

So I thought "what Amazon do ?" 

In fact the answer is quite simple : Amazon provides a list of all ereader devices or apps connected to the account, this list is available on the book details page.
![amazon-liste-destination-ebook-kindle-devices-and-apps](https://cloud.githubusercontent.com/assets/972023/24098445/b67f4024-0d6a-11e7-9cb0-a457dde9ab9a.PNG)

What I would like is for COPS to provide the same functionality 
a functionality that can be modified with the config section.

First a question would be asked
Do you want to use multiple emails ? (checkbox)

If checked, it would be possible to provide as many items as wished with a name and email.
default device : John's reader / email : john.doe@kindle.com
2nd device : Bob's reader / email : bob.doe@kindle.com
3rd device ...
and so on
You can add / modify / delete an entry

mockup
![mockup amelioration sendto-with list](https://cloud.githubusercontent.com/assets/972023/24098802/a9a06bf6-0d6c-11e7-925c-592b7a3829e6.PNG)

of course, there would be a : "other destination" and that would open a dialog box to enter the email address.

It would be a huge improvement to the functionality !
#326

**Comment by hugoheitz** [comment]
First : let's add the mail button !

It needs to add this part of the file :  cops/templates/default/bookdetail.html
        {{? data.mail == 1}}
        <a id="mailButton" title="Mail // {{=data.format}}" href="empty.php" onclick="sendToMailAddress (this, {{=data.id}}); return false;" {{=it.c.config.kindleHack}}><i class="icon-envelope icon-large"></i></a>
        {{?}}

to this file : 
 cops/templates/default/main.html

It should be possible to add a class as the existing class download below to add the mail button
proposed class 
           <h2 class="sendto">
        {{? data.mail == 1}}
        <a id="mailButton" title="Mail // {{=data.format}}" href="empty.php" onclick="sendToMailAddress (this, {{=data.id}}); return false;" {{=it.c.config.kindleHack}}><i class="icon-envelope icon-large"></i></a>
        {{?}}
            </h2>

existing class download
            <h2 class="download">
                {{~entry.book.preferedData:data:j}}
                 <a href="{{=data.url}}" {{=it.c.config.kindleHack}}>{{=data.name}}</a><br />
                {{~}}
            </h2>

Once the mail button will be available, we'll see how implementing with multiple devices (as it is not something that we use all the time).

**Comment by hugoheitz** [comment]
Second, let's use the config file to configure potential destination devices.
file : cops/config_default.php

Why I propose to use the config file ? 
Because it is not frequent that we acquire a new ereader, so the config file seems a correct place for it.
Also it is simplier because it does not need to have informations elsewhere.

sendto section of the config_default.php

     * If you want to use a simple smtp server (provided by your ISP for example), you can configure it like that :
     * $config['cops_mail_configuration'] = array( "smtp.host"     => "smtp.free.fr",
     *                                           "smtp.username" => "",
     *                                           "smtp.password" => "",
     *                                           "smtp.secure"   => "",
     *                                           "address.from"  => "cops@slucas.fr",
     *                                           "subject"       => "Sent by COPS : " // Not mandatory
     *                                           Device section of mail configuration
     *                                           to be used with kindle devices : you need to have a device and an email associated for each device.
     *                                           the device1 is the default device
     *                                           "device1_name"     => "name of your device1 kindle"
     *                                           "mail_device1"       => "destination mail of your device1 kindle"
     *                                           "device2_name"     => "name of your device2 kindle"
     *                                           "mail_device2"       => "destination mail of your device2 kindle"
     *                                          );

With a loop, it would be possible to populate the list that would be visible in the interface.

So the selection list would display the following items
device1               (selected by default)
device2
the mail chosen in the config page
other email         (it would then ask for email selection)
